qemu/tests/dump_guest_core.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import os
+
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import utils_package
+from virttest import utils_misc
+
+
+def run(test, params, env):
+    """
+    Test dump-guest-core, this case will:
+
+    1) Start VM with dump-guest-core option.
+    2) Check host env.
+    3) Trigger a core dump in host.
+    4) Use gdb to check core dump file.
+    5) If dump-guest-core=on, use crash to check vmcore file
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environmen.
+    """
+
+    def check_env():
+        """
+        Check if host kernel version is same with guest
+        """
+        guest_kernel_version = session.cmd("uname -r").strip()
+        logging.info("host kernel version is %s" % host_kernel_version)
+        logging.info("guest kernel version is %s" % guest_kernel_version)
+        if host_kernel_version != guest_kernel_version:
+            test.cancel("Please update your host and guest kernel "
+                        "to same version")
+
+    def check_core_file():
+        """
+        Use gdb to check core dump file
+        """
+        arch = 'X86_64'
+        arch = 'ppc64-le' if params['vm_arch_name'] == 'ppc64le' else arch
+        command = ('echo -e "source %s\nset height 0\ndump-guest-memory'
+                   ' %s %s\nbt\nquit" > %s' % (dump_guest_memory_file,
+                                               vmcore_file, arch,
+                                               gdb_command_file))
+        process.run(command, shell=True)
+        status, output = process.getstatusoutput(gdb_command, timeout=360)
+        os.remove(gdb_command_file)
+        os.remove(core_file)
+        logging.debug(output)
+        if status != 0:
+            test.fail("gdb command execute failed")
+        elif "<class 'gdb.MemoryError'>" in output:
+            if dump_guest_core == "on":
+                test.fail("Cannot access memory")
+
+    def check_vmcore_file():
+        """
+        Use crash to check vmcore file
+        """
+        process.run('echo -e "bt\ntask 0\ntask 1\nquit" > %s'
+                    % crash_script, shell=True)
+        crash_cmd = "crash -i %s /usr/lib/debug/lib/modules/%s/vmlinux %s"
+        crash_cmd %= (crash_script, host_kernel_version, vmcore_file)
+        status, output = process.getstatusoutput(crash_cmd, timeout=60)
```

Comment:
If status is useless here could be `output = process.getoutput(crash_cmd, timeout=60)`